Framework/DataObjects/test/ScanningWorkspaceBuilderTest.h

```diff
@@ -320,6 +328,58 @@ class ScanningWorkspaceBuilderTest : public CxxTest::TestSuite {
   }
 
   void
+  test_creating_workspace_with_relative_rotations_on_previously_rotated_detectors() {
+
+    const auto &instWS =
+        WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(
+            int(nDetectors), int(nBins));
+    auto &instDetInfo = instWS->mutableDetectorInfo();
+
+    Quat rotation = Quat(10.0, V3D(0, 1, 0));
```

Comment:
If both the original rotation and the updated one are around the same axis you are not catching a wrong rotation order (remember, rotation matrices to not commute!).